### DISCUSSION TITLE
Milvus , scaling Data insertion using distributed environment like Ray or Spark .

### DISCUSSION THREAD

**Discussion by ez-ml** [milvus-io/milvus#29821] [Q&A and General discussion]
I am looking for a way to scale the data ingestion in Milvus for huge amount of vector data in TB scale and looking for best approach to do this  ?.

I have tried using distributed cluster like Ray , but encountering the issue same as discussed in (https://github.com/milvus-io/milvus/discussions/21833) . 

Some example would be good to address this.

**Comment by yhmo** [comment]
https://github.com/milvus-io/milvus/discussions/21833#discussioncomment-7366684
Does this answer work for you?

**Comment by xiaofan-luan** [comment]
The best way we suggest is to use spark to prepare your data and do data ingestion.

We are gonna to release spark connector soon.

With the spark connector, you will be able to prepare data， write into storage and bulkinsert into milvus.
Estimated release date of spark milvus will be end of this month
